lib/server.js

```diff
@@ -259,6 +285,52 @@ module.exports = class Server extends EventEmitter {
       return hs
     }
 
+    if (relayThrough || remotePayload.relayThrough) {
+      let isInitiator
+      let publicKey
+      let token
+
+      if (relayThrough) {
+        isInitiator = true
+        publicKey = relayThrough
+        token = hs.relayToken
+      } else {
+        isInitiator = false
+        publicKey = remotePayload.relayThrough.publicKey
+        token = remotePayload.relayThrough.token
+      }
+
+      hs.relayToken = token
+      hs.relaySocket = this.dht.connect(publicKey)
+      hs.relayClient = relay.Client.from(hs.relaySocket, { id: hs.relaySocket.publicKey })
+
+      hs.relayClient
+        .pair(isInitiator, token, hs.rawStream)
+        .on('error', () => hs.relaySocket.destroy())
```

Comment:
you don't need to attach on(close) for this? (and ignore onerror) - it might close without an error but unsure what the flow is here

Reply:
The pairing request will emit an error if the pairing is cancelled, in which case we can immediately destroy the relay socket. Otherwise, if the pairing succeeds, we maintain the relay socket until the raw stream closes